src/libs/auth.ts

```diff
@@ -97,71 +127,206 @@ const revokeTokens = async () => {
+          },
+        }));
+        Ajax().Metrics.captureEvent(Events.user.authTokenLoad.success, {
+          authProvider: (user.profile as B2cIdTokenClaims).idp,
+          ...labelOldAuthToken(oldAuthTokenMetadata),
+          authTokenCreatedAt: labelTimestampMetric(authTokenCreatedAt),
+          authTokenExpiresAt: labelTimestampMetric(authTokenExpiresAt),
+          authTokenId,
+          refreshTokenId: authStore.get().refreshTokenMetadata.id,
+          refreshTokenCreatedAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.createdAt),
+          refreshTokenExpiresAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.expiresAt),
+          jwtExpiresAt: labelTimestampMetric(jwtExpiresAt),
+        });
+      },
+    ],
+    [
+      'expired',
+      () => {
+        Ajax().Metrics.captureEvent(Events.user.authTokenLoad.expired, {
+          ...labelOldAuthToken(oldAuthTokenMetadata),
+          refreshTokenId: authStore.get().refreshTokenMetadata.id,
+          refreshTokenCreatedAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.createdAt),
+          refreshTokenExpiresAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.expiresAt),
+        });
+      },
+    ],
+    [
+      'error',
+      () => {
+        Ajax().Metrics.captureEvent(Events.user.authTokenLoad.error, {
+          // we could potentially log the reason, but I don't know if that data is safe to log
+          ...labelOldAuthToken(oldAuthTokenMetadata),
+          refreshTokenId: authStore.get().refreshTokenMetadata.id,
+          refreshTokenCreatedAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.createdAt),
+          refreshTokenExpiresAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.expiresAt),
+        });
+      },
+    ]
+  );
+
+  return loadedAuthTokenState;
+};
+
+const tryLoadAuthToken = async (includeBillingScope = false, popUp = false): Promise<AuthTokenState> => {
+  const args: ExtraSigninRequestArgs = getSigninArgs(includeBillingScope);
+  const authInstance = getAuthInstance();
+  const loadedAuthTokenResponse: User | null | AuthTokenErrorState = await (popUp
+    ? // returns Promise<User | null>, attempts to use the refresh token to get a new authToken
+      authInstance.signinPopup(args)
+    : authInstance.signinSilent(args)
+  ).catch((e) => {
+    const errorMsg = 'An unexpected exception occurred while attempting to load auth credentials.';
+    console.error(errorMsg);
+    console.error(e);
+    const errorState: AuthTokenErrorState = {
+      errorMsg,
+      reason: e,
+      status: 'error',
+    };
+    return errorState;
   });
-  return user;
+  if (loadedAuthTokenResponse instanceof User) {
+    return {
+      status: 'success',
+      user: loadedAuthTokenResponse,
+    };
+  }
+  if (loadedAuthTokenResponse === null) {
+    return {
+      status: 'expired',
+    };
+  }
+
+  return loadedAuthTokenResponse;
```

Comment:
See above comment about the type of `loadedAuthTokenResponse`. This looks invalid, but also unreachable.